src/sentry/tasks/backfill_outboxes.py

```diff
@@ -0,0 +1,187 @@
+"""
+Checks OutboxProduciongModel classes and their __replication_version__.
+When the replication_version on any class is bumped, this task queues up
+outboxes for all the model instances of said class.
+
+This helps keep replicated models replicated when logic or columns change over time.
+"""
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import Any, Callable, Tuple, Type, Union
+
+import sentry_sdk
+from celery import Task
+from django.apps import apps
+from django.db import router, transaction
+from django.db.models import Max
+
+from sentry.db.models import BaseModel
+from sentry.db.models.outboxes import ControlOutboxProducingModel, RegionOutboxProducingModel
+from sentry.models import outbox_context
+from sentry.silo import SiloMode
+from sentry.tasks.base import instrumented_task
+from sentry.utils import json, metrics, redis
+
+
+@dataclass
+class BackfillBatch:
+    low: int
+    up: int
+    version: int
+    has_more: bool
+
+
+def get_backfill_key(table_name: str) -> str:
+    return f"outbox_backfill.{table_name}"
+
+
+def get_processing_state(table_name: str) -> Tuple[int, int]:
+    with redis.clusters.get("default").get_local_client_for_key("backfill_outboxes") as client:
+        key = get_backfill_key(table_name)
+        v = client.get(key)
+        if v is None:
+            result = (0, 0)
```

Comment:
If the default version number is 0 here, but 1 on the models, won't we trigger a backfill any time we lose state in redis?

Reply:
That's a solid point.  It's actually a fatal flaw of the deletion logic too.  But that puts us in the position of needing to implement "counter" tables in both region and control, which I didn't super like either.

Maybe that needs some more thought.